### PR TITLE
Add Dotenv and download .env from S3

### PR DESCRIPTION
**Pull request by zachmargolis**
**Why**: More flexible secret management

--

environment variables are not my favorite, but I really like the simplicity of the dotenv gem for stuff like this

Also used `#fetch` to be extra yelly

### CODE REVIEW

File: config/secrets.yml [hunk 1]
```diff
@@ -29,13 +29,13 @@ test:
...
-  saml_issuer: urn:gov:gsa:SAML:2.0.profiles:sp:sso:rails-<%= ENV["SAML_ENV"] %>
-  idp_sso_url: https://idp.<%= ENV["SAML_ENV"] %>.login.gov/api/saml/auth
-  idp_slo_url: https://idp.<%= ENV["SAML_ENV"] %>.login.gov/api/saml/logout
-  http_auth_username: <%= ENV["SP_NAME"] %>
-  http_auth_password: <%= ENV["SP_PASS"] %>
+  secret_key_base: <%= ENV.fetch("SECRET_KEY_BASE") %>
+  saml_issuer: urn:gov:gsa:SAML:2.0.profiles:sp:sso:rails-<%= LoginGov::Hostdata.env %>
+  idp_sso_url: https://idp.<%= LoginGov::Hostdata.env %>.<%= LoginGov::Hostdata.domain %>/api/saml/auth
+  idp_slo_url: https://idp.<%= LoginGov::Hostdata.env %>.<%= LoginGov::Hostdata.domain %>/api/saml/logout
+  http_auth_username: <%= ENV.fetch("SP_NAME") %>
```
Comment (brodygov) [inline config/secrets.yml]: Can we rename `SP_NAME` to something more descriptive while we're at it?

File: config/secrets.yml [hunk 2]
```diff
@@ -29,13 +29,13 @@ test:
...
 production:
-  secret_key_base: <%= ENV["SECRET_KEY_BASE"] %>
-  saml_issuer: urn:gov:gsa:SAML:2.0.profiles:sp:sso:rails-<%= ENV["SAML_ENV"] %>
-  idp_sso_url: https://idp.<%= ENV["SAML_ENV"] %>.login.gov/api/saml/auth
-  idp_slo_url: https://idp.<%= ENV["SAML_ENV"] %>.login.gov/api/saml/logout
-  http_auth_username: <%= ENV["SP_NAME"] %>
-  http_auth_password: <%= ENV["SP_PASS"] %>
+  secret_key_base: <%= ENV.fetch("SECRET_KEY_BASE") %>
+  saml_issuer: urn:gov:gsa:SAML:2.0.profiles:sp:sso:rails-<%= LoginGov::Hostdata.env %>
+  idp_sso_url: https://idp.<%= LoginGov::Hostdata.env %>.<%= LoginGov::Hostdata.domain %>/api/saml/auth
```
Comment (brodygov) [inline config/secrets.yml]: These URLs should be a little bit more templated since we use `"secure.login.gov"` when `Hostdata.env == 'prod'`.

**Comment by brodygov** [comment]
P.S. `SECRET_KEY_BASE` isn't set for circleci.

File: config/secrets.yml (hunk 2 above)
Comment (zachmargolis) [reply config/secrets.yml]: PTAL @ 2cf12b1a452e0fc9687286418d46a522a219e00b

File: config/secrets.yml (hunk 1 above)
Comment (zachmargolis) [reply config/secrets.yml]: Done!

File: config/secrets.yml [hunk 3]
```diff
@@ -29,13 +29,12 @@ test:
...
-  http_auth_username: <%= ENV["SP_NAME"] %>
-  http_auth_password: <%= ENV["SP_PASS"] %>
-  # 68:19:... is the fingerprint of the SAML IDP cert CN=int.login.gov
-  # Serial: 0xf41d6ee2e4675981, Issued: July 11 2017, Expires: July 11 2018
-  idp_cert_fingerprint: '68:19:E0:CB:D2:44:A0:CD:B7:D1:80:8A:E2:34:E4:22:90:41:DF:10'
-  acs_url: https://sp.<%= ENV["SAML_ENV"] %>.login.gov/auth/saml/callback
+  <% idp_host = LoginGov::Hostdata.env == 'prod' ? 'secure' : 'idp' %>
+  secret_key_base: <%= ENV.fetch("SECRET_KEY_BASE") %>
+  saml_issuer: urn:gov:gsa:SAML:2.0.profiles:sp:sso:rails-<%= LoginGov::Hostdata.env %>
+  idp_sso_url: https://<%= idp_host %>.<%= LoginGov::Hostdata.env %>.<%= LoginGov::Hostdata.domain %>/api/saml/auth
```
Comment (brodygov) [inline config/secrets.yml]: Still not quite right, this renders as `secure.prod.login.gov` rather than `secure.login.gov`.

Comment (zachmargolis) [reply config/secrets.yml]: Ah dangit, good catch

Comment (zachmargolis) [reply config/secrets.yml]: PTAL @ 24e89ca1f20192fe510ef3a8cf917e85d6ff0c9d

**Pull request closed without merging**